library/spdm_responder_lib/libspdm_rsp_certificate.c

```diff
@@ -104,46 +103,49 @@ libspdm_return_t libspdm_get_response_certificate(void *context,
                                                response_size, response);
     }
 
-    if (spdm_context->local_context
-        .local_cert_chain_provision[slot_id] == NULL) {
+    if (spdm_context->local_context.local_cert_chain_provision[slot_id] == NULL) {
         return libspdm_generate_error_response(
             spdm_context, SPDM_ERROR_CODE_INVALID_REQUEST,
             0, response_size, response);
     }
 
     offset = spdm_request->offset;
     length = spdm_request->length;
-    if (length > LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) {
-        length = LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN;
-    }
+
     if (length == 0) {
         return libspdm_generate_error_response(spdm_context,
                                                SPDM_ERROR_CODE_INVALID_REQUEST, 0,
                                                response_size, response);
     }
-
-    if (offset >= spdm_context->local_context
-        .local_cert_chain_provision_size[slot_id]) {
+    if (offset >= spdm_context->local_context.local_cert_chain_provision_size[slot_id]) {
         return libspdm_generate_error_response(spdm_context,
                                                SPDM_ERROR_CODE_INVALID_REQUEST, 0,
                                                response_size, response);
     }
 
-    libspdm_reset_message_buffer_via_request_code(spdm_context, session_info,
-                                                  spdm_request->header.request_response_code);
+    if (!libspdm_is_capabilities_flag_supported(spdm_context, false,
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP,
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP)) {
+        if (length > LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) {
+            length = LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN;
+        }
+    }
```

Comment:
Why you add check cap check for LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN ?

Reply:
Is there any test to cover the new case: (CHECK_CAP == 1) && (length > LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) ?

Reply:
> Why you add check cap check for LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN ?

If chunking is not supported then it reverts to the old behavior.

Reply:
I am thinking below:
1) If LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN is ignored in CHUNK, then we should describe in spdm_config.h.
2) If we want to change, we should also change in encap_certificate.
3) We need have new test to cover this new corner case, to ensure it still works in all scenario.

Reply:
Looking through the code I don't think `LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN` is being used correctly / consistently. For example `libspdm_get_certificate` uses `LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN` as the maximum size of the entire certificate, rather than just a portion. It's possible it's not needed anymore now that we have the send and receive buffers. I'll file a separate issue.

Reply:
> For example libspdm_get_certificate uses LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN as the maximum size of the entire certificate, rather than just a portion.

Would you please point out where the code is? I think length means portion_length
```
    length = LIBSPDM_MIN(length, LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN);
```
Why it is entire certificate?